provider/azure/environ.go

```diff
@@ -836,14 +869,18 @@ func deleteInstance(
 	// Delete the VM's OS disk VHD.
 	logger.Debugf("- deleting OS VHD")
 	blobClient := storageClient.GetBlobService()
-	if _, err := blobClient.DeleteBlobIfExists(osDiskVHDContainer, vmName); err != nil {
+	if err := callAPI(func() (autorest.Response, error) {
+		_, err := blobClient.DeleteBlobIfExists(osDiskVHDContainer, vmName)
```

Comment:
It's a different Azure API altogether. Actually there's no point that being wrapped in callAPI, I'll take it back out.